mailer/mailer.go

```diff
@@ -22,13 +27,17 @@ type MailSubjects struct {
 }
 
 // NewMailer returns a new authlify mailer
-func NewMailer(conf *conf.Configuration) *Mailer {
+func NewMailer(conf *conf.Configuration) Mailer {
 	mailConf := conf.Mailer
-	return &Mailer{
+	if mailConf.AdminEmail == "" || mailConf.Host == "" || mailConf.Port == 0 {
```

Comment:
personally, I'd prefer to make the `conf.Mailer` a pointer, then we can check for nil instead of checking internal state. Plus it signals that it is optional in the config object